libmscore/line.cpp

```diff
@@ -531,7 +531,7 @@ QPointF SLine::linePos(GripLine grip, System** sys) const
                         else if (type() == Element::Type::LYRICSLINE) {
                               // layout to right edge of CR
                               if (cr)
-                                    x = cr->width();
+                                    x = cr->pos().x() + cr->width();
```

Comment:
What happens with this code is that notes larger than a quarter note get treated very differently depending on whether they are "stems up" (or would have stems up if they had stems) versus "stems down".  Try it - end a melisma line on a whole note treble clef top space E, then watch what happens if you drop it down an octave.  The length of the melisma line changes.

The reason is that the chord's x offset is reversed for these two cases - it's positive in one case, negative in the other (and the notehead x offsets then counter this).  I had implemented these chord offsets some time ago when tweaking the layout code and the notehead-centering algorithms in particular.  I forget the exact reasons why the directions are reversed according to stem direction, but I know it solved some problem that would have been hard to solve another way - something to do with stem positions I think.

So, the lengths are nice and consistent if we *don't* try to incorporation the chord x position, but they start getting funny if we do, because of what I explained above.  The chord x offsets were designed to work together with the note x offsets; on their own, they don't tell the whole story.  *In theory*, the chord bbox() would have been already calculated and would give us the right answer, so we'd be able to use cr->bbox().right() here.  But unfortunately, chord bbox() just isn't reliable.  I don't understand what goes wrong with the calculation, but it's unpredictable - I get different answers at different times for the exact same chord.  I looked at where it is calculated and have no idea what the story is, so I decided to just do a brute force calculation here.